src/Spectre.AngularClient/src/app/divik-results/shared/divik.service.ts

```diff
@@ -24,53 +24,82 @@ import 'rxjs/Rx';
 
 import { Heatmap } from '../../heatmaps/shared/heatmap';
 import { Service } from '../../app.service';
+import { DivikConfig } from './divik-config';
 
 @Injectable()
 export class DivikService extends Service {
 
-    constructor(private http: Http) { super(); }
+  constructor(private http: Http) {
+    super();
+  }
 
-    private getHeaders() {
-      const headers = new Headers();
-      headers.append('Accept', 'application/json');
-      return headers;
-    }
+  private getHeaders() {
+    const headers = new Headers();
+    headers.append('Accept', 'application/json');
+    return headers;
+  }
 
-    get(preparationId: number, divikId: number, level: number): Observable<Heatmap> {
-      const queryUrl = `${this.getBaseUrl()}/divikResult/${preparationId}?divikId=${divikId}&level=${level}`;
-      const response = this.http.get(queryUrl, {headers: this.getHeaders()});
-      return response.map(toHeatmap);
-    }
+  get (preparationId: number, divikId: number, level: number): Observable<Heatmap> {
+    const queryUrl = `${this.getBaseUrl()}/divikResult/${preparationId}?divikId=${divikId}&level=${level}`;
+    const response = this.http.get(queryUrl, {headers: this.getHeaders()});
+    return response.map(toHeatmap);
+  }
+
+  getConfig(preparationId: number, divikId: number): Observable<DivikConfig> {
+    const queryUrl = `${this.getBaseUrl()}/divikResult/${preparationId}?divikId=${divikId}`;
+    const response = this.http.get(queryUrl, {headers: this.getHeaders()});
+    return response.map(toDivikConfig);
+  }
+}
+
+function toDivikConfig(response: Response): DivikConfig {
+  const json = response.json();
+  return <DivikConfig>({
+    maxK: json.MaxK,
+    level: json.Level,
+    usingLevels: json.UsingLevels,
+    amplitude: json.Amplitude,
+    variance: json.Variance,
+    percentSizeLimit: json.PercentSizeLimit,
+    featurePreservationLimit: json.FeaturePreservationLimit,
+    metric: json.Metric,
+    plottingPartitions: json.PlottingPartitions,
+    plottingPartitionsRecursively: json.PlottingPartitionsRecursively,
+    plottingDecomposition: json.PlottingDecomposition,
+    plottingDecompositionRecursively: json.PlottingDecompositionRecursively,
+    maxDecompositionComponents: json.MaxDecompositionComponents,
+  })
+    ;
 }
 
 function toHeatmap(response: Response): Heatmap {
```

Comment:
Can we extract this part to single place? It duplicated code (with respect to Heatmap component). Maybe some other service? It would be created with owner name (`DivikService`), to set proper tracking in logs.